File: frontend/campus_app/lib/models/student_statistics.dart
import 'package:json_annotation/json_annotation.dart';

part 'student_statistics.g.dart';

@JsonSerializable()
class StudentStatistics {
  final int totalStudents; // Toplam öğrenci sayısı
  final int activeStudents; // Aktif öğrenci sayısı
  final int inactiveStudents; // Pasif öğrenci sayısı
  final int deletedStudents; // Silinmiş öğrenci sayısı
  final Map<String, int> departmentDistribution; // Departman dağılımı
  final Map<String, int> facultyDistribution; // Fakülte dağılımı
  final Map<String, int> genderDistribution; // Cinsiyet dağılımı
  final Map<String, int> gradeDistribution; // Sınıf durumu

  StudentStatistics({
    required this.totalStudents,
    required this.activeStudents,
    required this.inactiveStudents,
    required this.deletedStudents,
    required this.departmentDistribution,
    required this.facultyDistribution,
    required this.genderDistribution,
    required this.gradeDistribution,
  });

  factory StudentStatistics.fromJson(Map<String, dynamic> json) => _$StudentStatisticsFromJson(json);
  Map<String, dynamic> toJson() => _$StudentStatisticsToJson(this);
} 

File: backend/Bingol-University-Social-Media-main/src/main/java/bingol/campus/student/core/response/StudentStatistics.java
package bingol.campus.student.core.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class StudentStatistics {

    private long totalStudents;
    private long activeStudents;
    private long inactiveStudents;
    private long deletedStudents;
    private Map<String, Long> departmentDistribution; // Departman dağılımı
    private Map<String, Long> facultyDistribution; // Fakülte dağılımı
    private Map<String, Long> genderDistribution; // Cinsiyet dağılımı
    private Map<String, Long> gradeDistribution; // Sınıf durumu

    // Getter ve Setter metodları
}
